pkg/reconciler/route/route.go

```diff
@@ -274,7 +274,11 @@ func (c *Reconciler) externalDomainTLS(ctx context.Context, host string, r *v1.R
 			tls = append(tls, resources.MakeIngressTLS(cert, sets.List(dnsNames)))
 		} else {
 			acmeChallenges = append(acmeChallenges, cert.Status.HTTP01Challenges...)
-			r.Status.MarkCertificateNotReady(cert)
+			if cert.IsFailed() {
+				r.Status.MarkCertificateProvisionFailed(cert)
```

Comment:
Can you add some unit tests to add coverage to these cases?

eg. we have table tests here: https://github.com/knative/serving/blob/main/pkg/reconciler/route/table_test.go

Reply:
Sure, will add cases for cert failure.